stellargraph/random.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import random
+import tensorflow as tf
+
+
+_sg_random = random.Random()
+
+
+def generate_seed(seed=None):
+    """
+    Convenience function to use the global seed by default if the provided seed is None.
+
+    Args:
+        seed (int, optional): seed value
+
+    Returns:
+        seed if not None, otherwise the global seed
+
+    """
+    if seed is None:
+        return _sg_random.randint(0, 2 ** 32 - 1)
```

Comment:
Why 32-bit seeds, not 64-bit ones, out of curiosity?

Reply:
It seems like numpy only supports 32 bit seeds https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.random.seed.html#numpy.random.seed

and the seeds we generate from here does get used for numpy seeds